src/libs/parsers/tests/formatDates.test.ts

```diff
@@ -8,86 +8,368 @@ import {
+    const result = getFormattedDates([
+      DECEMBER_14_2020.toISOString(),
+      DECEMBER_14_2020.toISOString(),
+    ])
+    expect(result).toEqual('14 décembre 2020')
+  })
+
+  it('should return a formatted string date of two dates with the same month', () => {
+    const result = getFormattedDates([
+      NOVEMBER_12_2020.toISOString(),
+      NOVEMBER_13_2020.toISOString(),
+    ])
+    expect(result).toEqual('les 12 et 13 novembre 2020')
+  })
+
+  it('should return a formatted string date of 1 dates if there is two same dates', () => {
+    const result = getFormattedDates([
+      NOVEMBER_20_2020_MORNING.toISOString(),
+      NOVEMBER_20_2020_EVENING.toISOString(),
+    ])
+    expect(result).toEqual('le 20 novembre 2020')
+  })
+
+  it('should return a formatted string date of 2 dates with the same day, smae month, but different years', () => {
+    const result = getFormattedDates([
+      NOVEMBER_13_2020.toISOString(),
+      NOVEMBER_13_2021.toISOString(),
+    ])
+    expect(result).toEqual('le 13 novembre 2020 et le 13 novembre 2021')
+  })
+
+  it('should return a formatted string date that includes three dates with different months', () => {
+    const result = getFormattedDates([
+      AUGUST_5_2021.toISOString(),
+      SEPTEMBER_5_2021.toISOString(),
+      OCTOBER_5_2021.toISOString(),
+    ])
+    expect(result).toEqual('le 5 août 2021, le 5 septembre 2021 et le 5 octobre 2021')
+  })
+
+  it('should return a formatted string date that includes three dates, with two dates in different months of the same year and one dates in a different month and in a different year', () => {
+    const result = getFormattedDates([
+      JANUARY_5_2021.toISOString(),
+      OCTOBER_5_2021.toISOString(),
+      FEBRUARY_2_2022.toISOString(),
+    ])
+    expect(result).toEqual('le 5 janvier 2021, le 5 octobre 2021 et le 2 février 2022')
+  })
+
+  it('should return a formatted string date that includes three dates, with two dates in the same month of the same year and one dates in a different month and in a different year', () => {
+    const result = getFormattedDates([
+      JANUARY_5_2021.toISOString(),
+      JANUARY_15_2021.toISOString(),
+      FEBRUARY_2_2022.toISOString(),
+    ])
+    expect(result).toEqual('les 5 et 15 janvier 2021 et le 2 février 2022')
+  })
+
+  it('should return a formatted string date that includes 4 dates in the same year, with 2 dates in different months and 2 dates in the same month', () => {
+    const result = getFormattedDates([
+      JANUARY_5_2021.toISOString(),
+      JANUARY_15_2021.toISOString(),
+      OCTOBER_5_2021.toISOString(),
+      SEPTEMBER_5_2021.toISOString(),
+    ])
+    expect(result).toEqual('les 5 et 15 janvier 2021, le 5 septembre 2021 et le 5 octobre 2021')
+  })
+
+  it('Should return a formatted string date that includes 4 dates, with 1 dates in a different month and 3 dates in the same month', () => {
```

Comment:
`should return a formatted string date including 4 dates, with 1 date of a different month and 3 dates in the same month`